platform/BufferedSerial.cpp

```diff
@@ -94,6 +94,18 @@ int BufferedSerial::sync()
     return 0;
 }
 
+void BufferedSerial::sigio(Callback<void()> func) {
```

Comment:
callback in critical section exposed to a public event hook? What is so critical here?